exporter/lokiexporter/next_exporter.go

```diff
@@ -0,0 +1,177 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lokiexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter"
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter/internal/third_party/loki/logproto"
+)
+
+type nextLokiExporter struct {
+	config   *Config
+	settings component.TelemetrySettings
+	client   *http.Client
+	wg       sync.WaitGroup
+}
+
+func newNextExporter(config *Config, settings component.TelemetrySettings) *nextLokiExporter {
+	settings.Logger.Info("using the new Loki exporter")
+
+	return &nextLokiExporter{
+		config:   config,
+		settings: settings,
+	}
+}
+
+func (l *nextLokiExporter) pushLogData(ctx context.Context, ld plog.Logs) error {
+	pushReq, _ := logDataToLoki(l.settings.Logger, ld)
```

Comment:
`numDroppedLogs` is skipped here, maybe we can log  it here or remove from return args of `logDataToLoki`.

Reply:
I would remove it; we can reinstate that once we can record metrics on the components.